v2/internal/reconcilers/arm/arm_client_cache.go

```diff
@@ -191,35 +194,49 @@ func (c *ARMClientCache) getARMClientFromSecret(secret *v1.Secret) (*armClient,
 
 func (c *ARMClientCache) newCredentialFromSecret(secret *v1.Secret, nsName types.NamespacedName) (azcore.TokenCredential, string, error) {
 	var errs []error
+	var err error
+	var credential azcore.TokenCredential
+
 	subscriptionID, ok := secret.Data[config.SubscriptionIDVar]
 	if !ok {
-		err := core.NewSecretNotFoundError(nsName, errors.Errorf("credential Secret %q does not contain key %q", nsName.String(), config.SubscriptionIDVar))
+		err = core.NewSecretNotFoundError(nsName, errors.Errorf("credential Secret %q does not contain key %q", nsName.String(), config.SubscriptionIDVar))
```

Comment:
Why switch to a shared `err` variable - these all look to be isolated cases where having independently decared single-write variables would make sense.

Reply:
Linter complained about `shadow declaration`